tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py

```diff
@@ -91,50 +90,32 @@ def test_assign_agent_to_a_group(agent_target, clean_environment):
             brief: Reset the wazuh log files at the start of the test. Remove all registered agents from master.
     assertions:
         - Verify that after registering the agent key file exists in all nodes.
-        - Verify that after registering and before receiving agent group info, it has the 'default' group assigned.
         - Verify that after registering and after receiving agent group info, it has the 'group_test' group assigned.
     expected_output:
         - The agent 'Agent_name' with ID 'Agent_id' belongs to groups: group_test."
     '''
-    replace = '\n' + '            "timeout_agent_groups": 40,' '\n' + '            "agent_group_start_delay": 300,\n'
-
-    # Add modify agent_group_start_delay
-    host_manager.add_block_to_file(host=test_infra_managers[0], path=f"{WAZUH_PATH}{path_cluster}",
-                                   after='"timeout_agent_info": 40,', before='"check_worker_lastkeepalive": 60,',
-                                   replace=replace)
-
     # Create new group
     host_manager.run_command(test_infra_managers[0], f"/var/ossec/bin/agent_groups -q -a -g {id_group}")
 
-    worker_ip = host_manager.run_command(agent_target, f'hostname -i')
+    worker_ip = host_manager.run_command(agent_target, 'hostname -i')
 
     # Modify ossec.conf in agent
     host_manager.add_block_to_file(host=test_infra_agents[0], path=f"{WAZUH_PATH}/etc/ossec.conf",
                                    after="</crypto_method>", before="</client>", replace=enrollment_group)
     host_manager.add_block_to_file(host=test_infra_agents[0], path=f"{WAZUH_PATH}/etc/ossec.conf",
                                    after="<address>", before="</address>", replace=worker_ip)
 
-    restart_cluster([test_infra_managers[0]] + test_infra_agents, host_manager)
-    time.sleep(timeout)
+    restart_cluster(test_infra_agents, host_manager)
+
+    # Wait to Agent-groups send full task to end due to race condition
+    time.sleep(agent_groups_send_full)
```

Comment:
Instead of waiting 60 seconds, find the related log to reduce the waiting time.

Reply:
Done in 6bb6279